### PR TITLE
Snapshot, GCP, & Azure support

### PR DESCRIPTION
**Pull request by ibolton336**
Add support for copy method selection in storage class selection table

### CODE REVIEW

**Comment by ibolton336** [comment]
Need to add support for cloud providers

**Comment by ibolton336** [comment]
Currently blocked on snapshots with GCP:

backupStorageLocation

# Please edit the object below. Lines beginning with a '#' will be ignored,
# and an empty file will abort the edit. If an error occurs while saving this file will be
# reopened with the relevant failures.
#
apiVersion: velero.io/v1
kind: BackupStorageLocation
metadata:
  creationTimestamp: 2019-10-28T15:16:51Z
  generateName: ian-bolton-migration-
  generation: 4
  labels:
    app.kubernetes.io/part-of: openshift-migration
    migplan: f6e2d79e-f995-11e9-bb43-0ae41d99ae0c
  name: ian-bolton-migration-vd776
  namespace: openshift-migration
  resourceVersion: "1358479"
  selfLink: /apis/velero.io/v1/namespaces/openshift-migration/backupstoragelocations/ian-bolton-migration-vd776
  uid: f79e6b6e-f995-11e9-a2cc-069fe0147c84
spec:
  config: null
  objectStorage:
    bucket: ian-bolton-migration
    prefix: velero
  provider: gcp
status:
  lastSyncedRevision: ""
  lastSyncedTime: 2019-10-28T15:19:39.899555779Z

File: src/app/cluster/components/AddEditClusterModal/AddEditClusterForm.tsx [hunk 1]
```diff
@@ -222,6 +263,14 @@ const AddEditClusterForm: any = withFormik({
       errors.token = 'Required';
     }
 
+    // if (!values.isAzure) {
```
Comment (eriknelson) [inline src/app/cluster/components/AddEditClusterModal/AddEditClusterForm.tsx]: Can we mark these with a TODO comment to be followed up in another PR?

File: src/app/storage/components/AddEditStorageModal/ProviderForms/GCPForm.tsx [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+const currentStatusFn = addEditStatusText(componentTypeStr);
+const addEditButtonTextFn = addEditButtonText(componentTypeStr);
+
+const valuesHaveUpdate = (values, currentStorage) => {
+    if (!currentStorage) { return true; }
+
+    const existingMigStorageName = currentStorage.MigStorage.metadata.name;
+    const existingGCPBucket = currentStorage.MigStorage.spec.backupStorageConfig.gcpBucket;
+    const existingGCPBlob = atob(currentStorage.Secret.data['gcp-credentials']);
+    // const existingAccessKeyId = atob(currentStorage.Secret.data['aws-access-key-id']);
```
Comment (eriknelson) [inline src/app/storage/components/AddEditStorageModal/ProviderForms/GCPForm.tsx]: Can we pull this

File: src/app/storage/components/AddEditStorageModal/ProviderForms/GCPForm.tsx [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+                </Box>
+            </Flex>
+
+        </Form>
+
+    );
+};
+
+
+// valuesHaveUpdate - returns true if the formik values hold values that differ
```
Comment (eriknelson) [inline src/app/storage/components/AddEditStorageModal/ProviderForms/GCPForm.tsx]: Should remove this comment
